src/orders/components/OrderHistory/OrderHistory.tsx

```diff
@@ -128,39 +197,167 @@ const OrderHistory = ({
                       onNoteUpdateLoading={onNoteUpdateLoading!}
                       relatedId={related.id}
                       id={id}
-                      date={date}
-                      user={user}
+                      date={<OrderHistoryDate date={date} />}
+                      actor={toActor(user, app)}
                       message={message}
                       key={id}
-                      app={app}
+                      eventData={event}
+                      isLastInGroup={isLastInGroup}
                     />
                   );
                 }
 
+                if (
+                  type === OrderEventsEnum.ORDER_MARKED_AS_PAID ||
+                  type === OrderEventsEnum.ORDER_FULLY_PAID
+                ) {
+                  const hasTransactionRef = !!event.transactionReference;
+                  const hasContent = hasTransactionRef || message;
+
+                  return (
+                    <TimelineEvent
+                      date={<OrderHistoryDate date={date} />}
+                      eventData={event}
+                      actor={toActor(user, app)}
+                      eventType={type}
+                      isLastInGroup={isLastInGroup}
+                      title={
+                        <Box display="flex" gap={1} alignItems="center" flexWrap="wrap">
+                          <FormattedMessage id="KpP/aW" defaultMessage="Order was marked as" />
+                          <Pill
+                            color="success"
+                            size="small"
+                            label={intl.formatMessage({
+                              id: "u/vOPu",
+                              defaultMessage: "Paid",
+                            })}
```

Comment:
suggestion: we should re-use existing translations strings / or add context within the message